presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -62,26 +89,34 @@
  */
 public class BingTileFunctions
 {
-    private static final int TILE_PIXELS = 256;
-    private static final double MAX_LATITUDE = 85.05112878;
-    private static final double MIN_LATITUDE = -85.05112878;
-    private static final double MIN_LONGITUDE = -180;
-    private static final double MAX_LONGITUDE = 180;
-    private static final double EARTH_RADIUS_KM = 6371.01;
     private static final int OPTIMIZED_TILING_MIN_ZOOM_LEVEL = 10;
     private static final Block EMPTY_TILE_ARRAY = BIGINT.createFixedSizeBlockBuilder(0).build();
 
-    private static final String LATITUDE_OUT_OF_RANGE = "Latitude must be between " + MIN_LATITUDE + " and " + MAX_LATITUDE;
-    private static final String LATITUDE_SPAN_OUT_OF_RANGE = String.format("Latitude span for the geometry must be in [%.2f, %.2f] range", MIN_LATITUDE, MAX_LATITUDE);
-    private static final String LONGITUDE_OUT_OF_RANGE = "Longitude must be between " + MIN_LONGITUDE + " and " + MAX_LONGITUDE;
-    private static final String LONGITUDE_SPAN_OUT_OF_RANGE = String.format("Longitude span for the geometry must be in [%.2f, %.2f] range", MIN_LONGITUDE, MAX_LONGITUDE);
-    private static final String QUAD_KEY_EMPTY = "QuadKey must not be empty string";
-    private static final String QUAD_KEY_TOO_LONG = "QuadKey must be " + MAX_ZOOM_LEVEL + " characters or less";
-    private static final String ZOOM_LEVEL_TOO_SMALL = "Zoom level must be > 0";
-    private static final String ZOOM_LEVEL_TOO_LARGE = "Zoom level must be <= " + MAX_ZOOM_LEVEL;
-
     private BingTileFunctions() {}
 
+    @Description("Encodes a Bing tile into a bigint")
+    @ScalarOperator(CAST)
+    @SqlType(StandardTypes.BIGINT)
+    public static long castToBigint(@SqlType(BingTileType.NAME) long tile)
+    {
+        return tile;
+    }
+
+    @Description("Decodes a Bing tile from a bigint")
+    @ScalarOperator(CAST)
+    @SqlType(BingTileType.NAME)
+    public static long castFromBigint(@SqlType(StandardTypes.BIGINT) long tile)
+    {
+        try {
+            BingTile.decode(tile);
+        }
+        catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)